tsdb/index/index.go

```diff
@@ -954,7 +966,11 @@ func (w *Writer) writePosting(name, value string, offs []uint32) error {
 	}
 
 	w.buf2.Reset()
-	w.buf2.PutBE32int(w.buf1.Len())
+	l := w.buf1.Len()
+	if l > math.MaxUint32 {
```

Comment:
You got a build error here since l is in and not uint

Reply:
Thanks for explanation, should be fixed. Also added comment.